SSKeychain/SSKeychainQuery.m

```diff
@@ -193,6 +203,14 @@ - (NSMutableDictionary *)query {
 		[dictionary setObject:self.accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
 	}
 #endif
+	
+#if SSKEYCHAIN_LEGACY_MODE_AVAILABLE
+	if (@available(macOS 10.15, *)) {
+		if (self.legacyKeychainMode == NO) {
+			[dictionary setObject:@(YES) forKey:(__bridge id)kSecUseDataProtectionKeychain];
+		}
+	}
+#endif
```

Comment:
The documentation indicates that it safe to use `kSecUseDataProtectionKeychain` on all platforms; it is ignored everywhere but iOS.

If we want to keep it conditional, though, this code could probably be simplified a bit, though:

```objc
#if SSKEYCHAIN_LEGACY_MODE_AVAILABLE
if (self.isLegacyModeAvailable) {
    BOOL useNewKeychain = !(self.legacyKeychainMode);
    [dictionary setObject: @(useNewKeychain) forKey:(__bridge id)kSecUseDataProectionKeychain];
}
#endif
```